File: 백준/Silver/2822. 점수 계산/점수 계산.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        List<scoreAndIndex> scoreAndIndex = new ArrayList<>();
        int sumOfTopFive = 0;

        for (int i = 0; i < 8; i++) {
            int score = sc.nextInt();

            scoreAndIndex.add(new scoreAndIndex(score, i + 1));
        }

        // 내림차순 정렬
        Collections.sort(scoreAndIndex);

        // 가장 높은 점수 5개의 번호를 저장할 배열
        int[] indexOfTopFive = new int[5];

        for (int i = 0; i < 5; i++) {
            sumOfTopFive = sumOfTopFive + scoreAndIndex.get(i).score;
            indexOfTopFive[i] = scoreAndIndex.get(i).index;
        }

        // 오름차순 정렬
        Arrays.sort(indexOfTopFive);

        System.out.println(sumOfTopFive);

        for (int i = 0; i < 5; i++) {
            System.out.print(indexOfTopFive[i] + " ");
        }
        sc.close();
    }
}

// 2가지 데이터를 저장할 클래스
class scoreAndIndex implements Comparable<scoreAndIndex> {
    public int score;
    public int index;

    // 객체를 정렬하는 것은
    // https://www.daleseo.com/java-comparable-comparator/ 참고

    public scoreAndIndex(int score, int index) {
        this.score = score;
        this.index = index;
    }

    public int getScore() {
        return this.score;
    }

    @Override
    public int compareTo(scoreAndIndex sNi) {
        return sNi.score - getScore();
    }
}

File: 백준/Silver/2822. 점수 계산/점수 계산.c
#include <stdio.h>
#include <stdlib.h>

int descending(const void* a, const void* b) {
    int num1 = *(int*)a;
    int num2 = *(int*)b;

    if (num1 < num2) return 1;  // 내림차순 정렬
    if (num1 > num2) return -1; // 내림차순 정렬
    return 0;
}

int ascending(const void* a, const void* b) {
    int num1 = *(int*)a;
    int num2 = *(int*)b;

    if (num1 < num2) return -1;  // 오름차순 정렬
    if (num1 > num2) return 1; // 오름차순 정렬
    return 0;
}

struct scoreAndIndex {
    int score;
    int index;
} scoreAndIndex;

int main()
{
    struct scoreAndIndex scoreAndIndex[8];
    int sumOfTopFive = 0;

    for (int i = 0; i < 8; i++) {
        scanf("%d", &scoreAndIndex[i].score);
        scoreAndIndex[i].index = i + 1;
    }

    // 내림차순 정렬
    qsort(scoreAndIndex, 8, sizeof(struct scoreAndIndex), descending);

    // 가장 높은 점수 5개의 번호를 저장할 배열
    int indexOfTopFive[5];

    for (int i = 0; i < 5; i++) {
        sumOfTopFive = sumOfTopFive + scoreAndIndex[i].score;
        indexOfTopFive[i] = scoreAndIndex[i].index;
    }

    // 오름차순 정렬
    qsort(indexOfTopFive, 5, sizeof(int), ascending);

    printf("%d\n", sumOfTopFive);

    for (int i = 0; i < 5; i++) {
        printf("%d ", indexOfTopFive[i]);
    }

    return 0;
}